api/v3/Case.php

```diff
@@ -60,58 +62,117 @@
  *   api result array
  */
 function civicrm_api3_case_create($params) {
+  _civicrm_api3_case_format_params($params);
 
-  if (!empty($params['id'])) {
-    return civicrm_api3_case_update($params);
+  if (empty($params['id'])) {
+    // Creating a new case, so make sure we have the necessary parameters
+    civicrm_api3_verify_mandatory($params, NULL, array(
+        'contact_id',
+        'subject',
+        array('case_type', 'case_type_id'),
+      )
+    );
   }
+  else {
+    // Update an existing case
+    // FIXME: Some of this logic should move to the BAO object?
+    // FIXME: Should we check if case with ID actually exists?
+    if (!isset($params['case_id']) && isset($params['id'])) {
+      $params['case_id'] = $params['id'];
+    }
 
-  civicrm_api3_verify_mandatory($params, NULL, array(
-    'contact_id',
-    'subject',
-    array('case_type', 'case_type_id'))
-  );
-  _civicrm_api3_case_format_params($params);
+    if (array_key_exists('creator_id', $params)) {
+      throw new API_Exception(ts('You cannot update creator id'));
```

Comment:
Tiny picky point: I believe ts() isn't generally used for API exceptions. Discussed at https://chat.civicrm.org/civicrm/pl/qiotz5x6p7fxdegbiece7xhhmc . ts() isn't used for the other exceptions in this file, apart from an identical instance in civicrm_api3_case_update() which presumably this was copied from.